### PR TITLE
really fast camelcase

### PR DESCRIPTION
**Pull request by achille-roussel**
@segmentio/infra 
@sperand-io 

Follow up of https://github.com/segmentio/go-snakecase/pull/1

Before:

```
$ go test -bench .
PASS
BenchmarkCamelcase-4      500000          2490 ns/op
ok      github.com/segmentio/go-camelcase   1.280s
```

After:

```
$ go test -bench .
PASS
BenchmarkCamelcase-4    10000000           192 ns/op
ok      github.com/segmentio/go-camelcase   2.126s
```

==> 12x 💃

### CODE REVIEW

**Comment by f2prateek** [comment]
LGTM!

File: camel.go [hunk 1]
```diff
@@ -83,3 +89,17 @@ func isLower(c byte) bool {
 func isDigit(c byte) bool {
 	return c >= '0' && c <= '9'
 }
+
+func toLower(c byte) byte {
+	if isUpper(c) {
+		return c + ('a' - 'A')
```
Comment (f2prateek) [inline camel.go:95]: Does this get inlined? Is it faster if we replace `'a' - 'A'` with 26?

Comment (achille-roussel) [inline camel.go:95]: I'm 99.99% sure the compiler sees `'a' - 'A'` as a constant expression, because it can be expressed as `const a = 'a' - 'A'`.

Comment (achille-roussel) [inline camel.go:95]: I just tested with declaring this as a constant:

```
$ go test -bench .
PASS
BenchmarkCamelcase-4    10000000           193 ns/op
ok      github.com/segmentio/go-camelcase   2.147s
```

same results... ;)

Comment (f2prateek) [inline camel.go:95]: Sweet!

**Pull request merged**